tests/20-SideMenu/BoardsMenuSection.test.tsx

```diff
@@ -26,62 +27,77 @@ describe("Regular", () => {
+    boards.forEach((board, i) => {
+      expect(board).toHaveTextContent(`!${(Regular.args as BoardsSectionProps).boards[i].slug}`);
+    });
   });
   
-  //Not sure if this should be it's own test or part of the previous test
   test("Board menu items link to boards", async () => {
     render(<Regular />);
-  
-    //TODO: fill this
+    const actionReturn = jest.fn();
+    mocked(action).mockReturnValue(actionReturn);
+    
+    const boards = screen.getAllByRole("link");
+    for (const board of boards) {
+      fireEvent.click(board);
+      await waitFor(() => {
+        expect(action).toBeCalledWith("#slug");
+      });
+    };
   });
   
   test("Correctly renders board without updates", async () => {
     render(<Regular />);
-
-    //Not sure .toBe is the right thing to expect here, but can't try it to figure it out because Testing Library can't find my aria labels
-    expect(screen.getByText("!oncie-den")).toBe(screen.getByRole("link", { name: "oncie-den"}));
+    
+    expect(screen.getByRole("link", { name: "oncie-den"})).toHaveTextContent("!oncie-den");
   });
   
   test("Correctly marks boards with updates", async () => {
     render(<Regular />);
   
-    expect(screen.getByLabelText("gore has new updates", {exact:false}));
+    expect(screen.getByLabelText("gore has new updates")).toHaveTextContent("!gore");
   });
   
   test("Correctly marks outdated boards with updates", async () => {
     render(<Regular />);
   
-    //TODO: fill this
+    expect(screen.getByLabelText("crack has updates")).toHaveTextContent("!crack");
   });
   
   test("Correctly marks current board", async () => {
     render(<Regular />);
   
-    //TODO: fill this
+    expect(screen.getByRole("link", { current: "page" })).toHaveTextContent("!kink-memes");
   });
 
   test("Correctly marks muted board", async () => {
     render(<Regular />);
   
-    //TODO: fill this
+    expect(screen.getByLabelText("anime muted")).toHaveTextContent("!anime");
   });
 });
 
 describe("Empty", () => {
   test("Renders empty section", async () => {
     render(<Empty />);
     
-    // This gives error "Property 'emptyDescription' does not exist on type 'Partial<Partial<BoardsMenuSectionProps>>'." and I don't understand why???
-    // expect(screen.getByText(Empty.args!.emptyDescription))
+    // This might be overkill?
+    expect(screen.getByText((Empty.args as EmptySectionProps).title)).toBeVisible;
+    expect(screen.getByText((Empty.args as EmptySectionProps).emptyTitle)).toBeVisible;
+    expect(screen.getByText((Empty.args as EmptySectionProps).emptyDescription)).toBeVisible;
   });
 });
 
 describe("Loading", () => {
   test("Renders loading section", async () => {
     render(<Loading />);
-  
-    //TODO: fill this
+    
+    expect(screen.getByText((Loading.args as LoadingSectionProps).title)).toBeVisible;
+    expect(screen.getAllByLabelText("loading board link")).toHaveLength((Loading.args as LoadingSectionProps).placeholdersCount);
```

Comment:
As I was mentioning, I think announcing "loading board link" 4 times wouldn't be ideal (also, it's not a link since you can't click on it). I would add the aria-label to the container.

For the actual loading placeholders, i was thinking they can have role `progressbar` (https://www.w3.org/TR/wai-aria-1.1/#progressbar) with min=0 and max=100, but no value.

Reply:
My understanding was that the container having aria-busy would mean that screen readers generally won't read out any of the info on the descendants while it's true, so I was less concerned about the reading four times issue, but I can change them to have the progressbar role instead. I had some rational for not, but i don't remember what it was now.

Reply:
Oh then if that's how it goes, it sounds good! I would just change "loading board link" to "loading board placeholder" since there's no actual link.

Reply:
will do